src/components/services/Nextcloud.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+  <Generic :item="item">
+    <template #content>
+      <p class="title is-4">{{ item.name }}</p>
+      <p class="subtitle is-6">
+        <template v-if="versionstring">
+          Version {{ versionstring }}
+        </template>
+      </p>
+    </template>
+    <template #indicator>
+      <div v-if="status" class="status" :class="status">
+        {{ status }}
+      </div>
+    </template>
+  </Generic>
+</template>
+
+<script>
+import service from "@/mixins/service.js";
+import Generic from "./Generic.vue";
+
+export default {
+  name: "Nextcloud",
+  mixins: [service],
+  props: {
+    item: Object,
+  },
+  components: {
+    Generic,
+  },
+  data: () => ({
+    status: null,
+    fetchOk: null,
+    versionstring: null,
+    maintenance: null,
+  }),
+  computed: {
+    status: function () {
+      if (this.fetchOk) {
+        if (this.maintenance) {
+          return "maintenance";
+        } else {
+          return "online";
+        }
+      } else {
+        return "offline";
+      }
```

Comment:
I think this part could be simplified like this: 
```suggestion
      if (!this.fetchOk) {
        return "offline";
      }
      return this.maintenance ? "maintenance" : "online";
```